client/client.go

```diff
@@ -246,13 +233,16 @@ func getUrlForResource(base string, version string, resourceName string, subComm
 			namedParameter
 		return url, nil
 	}
+	var queryParams = ""
```

Comment:
why not to use https://golang.org/pkg/net/url/ for query constructing? actually it's a question to the entire module

Reply:
Actually just because of historical reasons when we first started testing it was much more simple, but I had some problems with the URL. I was working on while Alessandro was refactoring it at the same time. It was behaving differently depending on the parameter name.
Resty has it own builder you just need to pass the map of parameters.

Ex:
resp, err := resty.R().
      SetQueryParams(map[string]string{
          "page_no": "1",
          "limit": "20",
          "sort":"name",
          "order": "asc",
          "random":strconv.FormatInt(time.Now().Unix(), 10),
      }).
      SetHeader("Accept", "application/json").
      SetAuthToken("BC594900518B4F7EAC75BD37F019E08FBC594900518B4F7EAC75BD37F019E08F").
      Get("/search_result")